alphadia/workflow/optimization.py

```diff
@@ -822,9 +848,10 @@ def update_with_fdr(self, precursor_df: pd.DataFrame):
             The precursor dataframe for the current batch of the optimization lock (from workflow.perform_fdr).
         """
 
-        self.count = len(precursor_df[precursor_df["qval"] < 0.01])
-
-        self.has_target_num_precursors = self.count >= self.target_count
+        self._precursor_at_fdr_count = np.sum(precursor_df["qval"] < 0.01)
```

Comment:
Should the precursor_df be filtered here to include only targets and only one channel, e.g. ``np.sum((precursor_df["qval"] < 0.01) & (precursor_df["decoy"] == 0) & (precursor_df["channel"] == channel)
)``?
Alternatively it could perhaps be filtered in the peptidecentric.py file.

Reply:
added the filter